crates/bevy_ecs/src/removal_detection.rs

```diff
@@ -19,14 +19,25 @@ use std::{
     option,
 };
 
-/// Wrapper around a [`ManualEventReader<Entity>`] so that we
+/// Wrapper around [`Entity`] for [`RemovedComponents`].
+/// Internally, `RemovedComponents` uses these as an `Events<RemovedComponentEntity>`.
+#[derive(Debug, Clone)]
+pub struct RemovedComponentEntity(Entity);
```

Comment:
Could the `Entity` field be made `pub`? I would much rather use `event.0` over `event.into()`